### DISCUSSION TITLE
Failed to fetch token for your AAD app, please make sure you have configured your application correctly

### DISCUSSION THREAD

**Discussion by ladparth** [microsoft/Microsoft-Fabric-workload-development-sample#74] [Help] [answered]
Followed [Authentication Tutorial](https://learn.microsoft.com/en-us/fabric/workload-development-kit/authentication-tutorial) yet getting this error. I was able authenticate with consent.

![image](https://github.com/user-attachments/assets/7eb334a5-2a14-4772-b2ec-909a5d3271e1)

![image](https://github.com/user-attachments/assets/e9dd69fe-74c2-4bcb-8ed5-78a882d1e234)

**Comment by ramizsha** [comment]
Hi @ladparth, did you configure .env.dev file in the frontend folder?
Also, couple of things to double check:
1. The format of your audience is as mentioned in the docs.
2. The redirect URI starts with "http://localhost".

**Reply by ramizsha** [reply]
Another thing to check is that you have configured the right preauthorizations for your application.
https://learn.microsoft.com/en-us/fabric/workload-development-kit/authentication-tutorial#add-scopes-for-data-plane-api

**Reply by ladparth** [reply]
Hi @ramizsha, thanks a lot for your response. Yes i followed all steps as mentioned in the docs with one change. Instead of using Fabric.WorkloadSample, i have used Org.WorkloadSample as per some other discussions i read here. 

Also, i have added all necessary scopes in the entra app and updated env.dev file too.

Lastly, i also got a authentication popup and was able to grant consent but got this error.

**Reply by ramizsha** [reply, marked as answer]
Just to make sure, have you also configured the preauthroization for "871c010f-5e61-4fb1-83ac-98610a7e9110"?

**Reply by ladparth** [reply]
This was the issue. I missed that step. Thanks a lot @ramizsha 
I can't create an item yet. i am yet to setup backend, will it fix the error once setup is done?

![image](https://github.com/user-attachments/assets/2c409659-404b-4db6-a56d-8cfd20b8a8f0)

**Reply by ramizsha** [reply]
You need to set up your backend before being able to create an item - yes.
